base/abstractarray.jl

```diff
@@ -2609,28 +2609,35 @@ function _typed_hvncat_shape(::Type{T}, shape::NTuple{N, Tuple}, row_first, as::
     return A
 end
 
-function hvncat_fill!(A::AbstractArray{T, N}, scratch1::Vector{Int}, scratch2::Vector{Int}, d1::Int, d2::Int, as::Tuple{Vararg}) where {T, N}
+function hvncat_fill!(A::AbstractArray{T, N}, scratch1::Vector{Int}, scratch2::Vector{Int},
+                              d1::Int, d2::Int, as::Tuple) where {T, N}
+    length(scratch1) == length(scratch2) == N ||
+        throw(ArgumentError("scratch vectors must have as many elements as the destination array has dimensions"))
+    0 < d1 < 3 &&
```

Comment:
```suggestion
    0 < d1 <= N &&
```
?
Otherwise the inbounds below is out of bounds

Reply:
d1 has to be 1 or 2, and this function is only called for N > 2. Dispatch takes care of that, though I can add a check to ensure `N > 2` to be extra safe.